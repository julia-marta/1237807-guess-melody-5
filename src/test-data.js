import React from 'react';
import PropTypes from "prop-types";

export const noop = () => {};

export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [{
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `blues`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `jazz`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [{
      picture: `https://api.adorable.io/avatars/128/0`,
      artist: `John Snow`,
    }, {
      picture: `https://api.adorable.io/avatars/128/1`,
      artist: `Jack Daniels`,
    }, {
      picture: `https://api.adorable.io/avatars/128/2`,
      artist: `Jim Beam`,
    }],
  }
];

export const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export const SimpleMockComponent = () => <div />;
