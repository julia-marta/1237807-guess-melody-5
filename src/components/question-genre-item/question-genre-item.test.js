import React from 'react';
import renderer from 'react-test-renderer';
import QuestionGenreItem from "./question-genre-item";
import {noop, questions} from "../../test-data";

const answer = questions[0].answers[0];

it(`should QuestionGenreItem render correctly`, () => {
  const tree = renderer
    .create(<QuestionGenreItem answer={answer} id={0} onChange={noop} onPlayButtonClick={noop} isPlaying={false} userAnswer={true} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
