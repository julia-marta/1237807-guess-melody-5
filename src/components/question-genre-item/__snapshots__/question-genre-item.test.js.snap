// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should QuestionGenreItem render correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
    />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={true}
      className="game__input visually-hidden"
      id="answer-0"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-0"
    />
    <label
      className="game__check"
      htmlFor="answer-0"
    >
      Отметить
    </label>
  </div>
</div>
`;
